registry/pluginloader/loaders_18_test.go

```diff
@@ -0,0 +1,75 @@
+// +build go1.8,!race
+
+package pluginloader
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	check "gopkg.in/check.v1"
+
+	"github.com/docker/distribution/registry/auth"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+	"github.com/docker/distribution/registry/storage/driver/testsuites"
+)
+
+func TestPlugins(t *testing.T) {
+	tempdir, err := ioutil.TempDir("", "ditributiontestplugins")
+	if err != nil {
+		t.Fatalf("can not create tempdirectory %v\n", err)
+	}
+	defer os.RemoveAll(tempdir)
+
+	pluginpath := filepath.Join(tempdir, "testplugin"+suffix)
+	t.Logf("compile plugin into %s\n", pluginpath)
+	cmd := exec.Command("go", "build", "-o", pluginpath, "-buildmode", "plugin", "github.com/docker/distribution/testutil/testplugin")
```

Comment:
I like the pattern of doing an `exec.LookPath("go")` before and skipping if the go binary is not found. Unlikely in this case, but perhaps we can expand that to skip if `go version` does not contain `go1.8.` or later.